### ISSUE TITLE
Non-Recursive Preprocessor Macros are not expanded

### ISSUE THREAD

**Issue by BetsyMcPhail** [doxygen/doxygen#8119]
**Describe the bug**
Preprocessor macros that use the non-recursive (`:=`) format are not expanded.

**Screenshots**
Expected:

![Image 10-19-20 at 8 03 PM](https://user-images.githubusercontent.com/8505893/96524414-3324ea00-1246-11eb-8a93-6eb811c51d59.jpeg)

After 1.8.17:
![Image 10-19-20 at 8 04 PM](https://user-images.githubusercontent.com/8505893/96524468-5f406b00-1246-11eb-866d-23f6fd4bfebc.jpeg)

**To Reproduce**

Doxygen settings:
```
ENABLE_PREPROCESSING   = YES
MACRO_EXPANSION             = YES
EXPAND_ONLY_PREDEF        = YES
PREDEFINED                           = DRAKE_DEFAULT_COPY_AND_MOVE_AND_ASSIGN(X):=" \
/** \name Implements CopyConstructible, CopyAssignable, MoveConstructible, MoveAssignable */ \
/** \{ */ \
X(const X&) = default; \
X& operator=(const X&) = default; \
X(X&&) = default; \
X& operator=(X&&) = default; \
/** \} */"
```

Header file:
```
class ExpressionAddFactory {
 public:
  DRAKE_DEFAULT_COPY_AND_MOVE_AND_ASSIGN(ExpressionAddFactory)

  ExpressionAddFactory() = default;
};
```

**Version**
This bug was introduced in commit dbd8ceb (version 1.8.17).

Although similar to #7527, the fix to that issue did not fix this issue, it is still reproducible on master.

**Comment by albert-github** [comment]
Code (#8120) has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
